Analysis/DataModel/include/AnalysisDataModel/HFSecondaryVertex.h

```diff
@@ -204,6 +204,31 @@ auto CosThetaStarD0bar(const T& candidate)
 {
   return candidate.cosThetaStar(array{RecoDecay::getMassPDG(kKPlus), RecoDecay::getMassPDG(kPiPlus)}, RecoDecay::getMassPDG(421), 0);
 }
+
+// Jpsi → e+e-
+template <typename T>
+auto CtJpsi(const T& candidate)
+{
+  return candidate.ct(RecoDecay::getMassPDG(443));
+}
+
+template <typename T>
+auto YJpsi(const T& candidate)
+{
+  return candidate.y(RecoDecay::getMassPDG(443));
+}
+
+template <typename T>
+auto EJpsi(const T& candidate)
+{
+  return candidate.e(RecoDecay::getMassPDG(443));
+}
+
+template <typename T>
+auto InvMassJpsi(const T& candidate)
```

Comment:
this can also have a name change since its decay specific

Reply:
@vkucera what do you think? I also dont want to get carried away too much

Reply:
I agree.
Whenever multiple decay channels are foreseen, the code that is specific to one channel should be clearly identified as such.